src/components/PostsList.tsx

```diff
@@ -1,54 +1,54 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-
 import classNames from 'classnames';
 import React from 'react';
-import { Post } from '../types/Post';
-
-type Props = {
-  posts: Post[];
-  selectedPostId?: number;
-  onPostSelected: (post: Post | null) => void;
-};
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import * as selectedPostActions from '../features/selectedPost';
 
-export const PostsList: React.FC<Props> = ({
-  posts,
-  selectedPostId = 0,
-  onPostSelected,
-}) => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
+export const PostsList: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { selectedPost } = useAppSelector(state => state.selectedPost);
```

Comment:
Verify that the `state.selectedPost` structure matches the actual Redux state structure. If the state does not have a `selectedPost` property, this will cause an error.